.pipelines/templates/template-clean-subscription.yml

```diff
@@ -0,0 +1,52 @@
+parameters:
+  - name: purgeTTL
+    displayName: 'Timeout before resources are force cleaned.'
+    type: string
+    default: '6h'
+  - name: purgeCreatedTag
+    displayName: 'Name of the tag used in subscription marking created at'
+    type: string
+  - name: subscriptionCredentials
+    displayName: 'Subscription properties required to login'
+    type: string
+    default: ''
+  - name: subscriptionId
+    displayName: 'SubscriptionId to clean'
+    type: string
+    default: ''
+  - name: workingDirectory
+    type: string
+    default: ''
+  - name: dryRun
+    type: bool
+    default: false
+
+jobs:
+- job: Clean_E2E
+  pool:
+    vmImage: ubuntu-latest
+
+  steps:
+  - template: ./template-setup-golang-env.yml
+    parameters:
+      gobin: ${{ variables.GOBIN }}
+      gopath: ${{ variables.GOPATH }}
+      goroot: ${{ variables.GOROOT }}
+      modulePath: ${{ variables.modulePath }}
+
+  - script: |
```

Comment:
```
  steps:
  - template: ./template-setup-golang-env.yml
    parameters:
      gobin: ${{ variables.GOBIN }}
      gopath: ${{ variables.GOPATH }}
      goroot: ${{ variables.GOROOT }}
      modulePath: ${{ variables.modulePath }}

  - script: |
      set -e
      echo ${{ parameters.subscriptionCredentials }} | base64 -d -w 0 > spn.json
      export AZURE_CLIENT_ID=$(cat spn.json | jq -r '.clientId')
      export AZURE_CLIENT_SECRET=$(cat spn.json | jq -r '.clientSecret')
      export AZURE_TENANT_ID=$(cat spn.json | jq -r '.tenantId')
      export AZURE_SUBSCRIPTION_ID=$(parameters.subscriptionId)
      rm spn.json

      export AZURE_PURGE_RG_DEL_PREFIX="${{ parameters.resourceGroupDeletePrefix }}""
      export AZURE_PURGE_TTL="${{ parameters.purgeTTL }}"

      go run ./hack/clean.go --dryRun="${{ parameters.dryRun }}"
    workingDirectory: ${{ parameters.workingDirectory }}
    displayName: 🧹 Clean E2E subscription
```
And lets remove anything which "We not gonna change in next 2 months, or EVER" from configurable.

Reply:
So we keep only `clean-subscription.yml` and `template-clean-subscription.yml`